### ISSUE TITLE
Add APKs as artifacts in PRs

### ISSUE THREAD

**Issue by maelchiotti** [maelchiotti/LocalMaterialNotes#44]

**Comment by maelchiotti** [comment]
@itsmartashub APKs are now available as artifacts in PRs, so you can try the [next release ](https://github.com/maelchiotti/LocalMaterialNotes/pull/41) right now if you wish!

**Comment by itsmartashub** [comment]
Hey, really thank you for creating the workflow with the build artifact so we can track the app's progress while we impatiently await the new version. 🙏

I hope that this probably too long feedback will be acceptable to you. These are just some subjective suggestions while trying out the workflow version ([`v1.3.0`](https://github.com/maelchiotti/LocalMaterialNotes/actions/runs/8938731847) dev PR) with grid view and rich text implemented. I'm sorry if this is too much for the current state of the app.

### GRID VIEW 
First of all, thank you so much for putting effort into implementing the grid view, and for doing it fairly quickly. If you agree with any of the following suggestions, maybe the view could be further improved by adding additional style:

- **Material You background color for notes in notes view**. However, this might be problematic for you as a developer in cases where the note is one color while in notes view, and another when we open the note in edit mode. This might not be a problem for light or even dark themes, but for example, in the OLED theme, the note would be Material You color in notes view, and when we open the note in edit mode, it should still be pure black. If you understand what I mean, huh
- On the other hand, maybe someone would prefer the note not to have a background but perhaps only an outline, so customization where the user could choose this would be a hit, but I totally understand that this might be too much and not a priority at the moment.
- If a background note were added, they would probably look sticked together (like when they are selected and stuck together), so maybe it would be appropriate to add some **small horizontal and vertical gap between notes**.
- **Rounded edges** would contribute to a more modern look in this case.
- If it suits you this way, then great, but just a small suggestion, maybe **increase the max-height** for notes in notes view a bit.

This would significantly improve the visual display and clear separation of one note from another (especialy in grid view where there are not separators).

### DELETING NOTES

- **Deleted note isn't removing from the UI at first**: By the way, I'm not sure if this is a bug or a feature, but when we select one or more notes while in notes view and delete them, they are not immediately visually removed from the UI (moved to trash), but only after we start writing a new note or enter an existing one and then return to notes view. Then the deleted notes are removed from the UI. If you intentionally left it this way, then maybe it wouldn't be a bad idea to display some toast message or some form of feedback.

- **Permanent note deletion not working**: Also, when mentioning deleting notes, I don't know if I'm doing something wrong, but it seems like permanent note deletion doesn't work. Whether I try to delete them one by one from the trash or if I select multiple ones, or if I do it directly from the edit mode.


### SYSTEM NAVIGATION (GESTURES OR BUTTONS)

- Also, I'm not sure if this is intentional, but when we go back from `Settings`, we exit the application instead of maybe returning to notes view. But maybe this is just a subjective habit.

### RICH TEXT EDITOR

- Markdown exporting or sharing to applications such as maybe Obsidian (markdown app), **cannot read a link from the note**. Also, it seems like **notes don't look very markdown-ish** when exported or shared to markdown applications.
- It's probably just me since I'm not into the rich text editor but markdown rather, but I'm not sure what the 5th (`Tt` underlined) and 6th (✏️) icon/tools should do?
- Is there a way for rich text tools to be customizable, like is there a way to change the order of some tools, for example linking since it is used often?

If you need a screenshot or recording to illustrate anything above, please let me know.

**Comment by maelchiotti** [comment]
Thank you for this amazing feedback @itsmartashub. I will make sure to look into it and provide an answer to your suggestions!

**Comment by maelchiotti** [comment]
@itsmartashub Here are answers to your feedback:

- **Grid view**: I don't think using a background for the notes tiles is a great idea, because it creates a very colorful app, and I prefer the plain look it has now.
- **Navigation**: Although the Settings page isn't technically on lower "level", I agree that it's more intuitive to go back to the previous page when exiting it, so I've fixed this #54.
- **Rich text editor**:
  - As you've noted, markdown exporting is lacking now that I added the rich text editor. I'll improve it in #56, but later, because it requires "parsing" the content. I've already done it to add the checkboxes in the preview of the first three lines of a note though, so I know how to proceed.
  - I've removed the buttons you talked about that are for the text color, but I won't support them for now.
  - As for ordering the buttons, it's not supported by [the package](https://github.com/fleather-editor/fleather) is use. I could create my own toolbar or create an issue in their repository, I'll think about it (#57).
- **Bugs**: the UI not updating and the bin not emptying are fixed.

I thank you again for the time taken to write your feedback!
